File: app/src/main/java/com/noteapp/Controll/NewNotesController.java
package com.noteapp.Controll;

import android.content.Context;
import android.view.View;
import android.widget.EditText;

import com.noteapp.Model.Note;
import com.noteapp.Model.NoteDatabase;
import com.noteapp.View.MainActivity;

import java.util.Date;

public class NewNotesController {

    private Context mContext;
    private NoteDatabase mDatabase;
    private MainController mainController;

    public NewNotesController(Context context) {
        this.mContext = context;
        this.mainController = MainActivity.mainController;
        this.mDatabase = mainController.provideDatabase();
    }


    private void saveNote(String title, String text, Date date) {
        mDatabase.insertNote(new Note(0, title, text, date));
    }

    public View.OnClickListener getAddBtnListener(final EditText mTitleET, final EditText mTextET) {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String title, text;
                Date date;

                title = mTitleET.getText().toString();
                text = mTextET.getText().toString();
                date = new Date();

                saveNote(title, text, date);
                mainController.updateNotesAdapterList();
            }
        };
    }
}

File: app/src/main/java/com/noteapp/Controll/NewNotesController.kt
package com.noteapp.Controll

import android.content.Context
import android.view.View
import android.widget.EditText
import android.widget.Toast
import com.noteapp.Model.Note
import com.noteapp.View.MainActivity
import java.util.*

class NewNotesController(val mContext: Context) {

    val mainController: MainController = MainActivity.mainController

    fun getAddBtnListener(mTitleET: EditText, mTextET: EditText): View.OnClickListener {
        return View.OnClickListener {
            val title: String = mTitleET.text.toString()
            val text: String = mTextET.text.toString()
            val date: Date = Date()

            mainController.saveNote(Note(0, title, text, date))
            mainController.updateNotesAdapterList()
            Toast.makeText(mContext, "New Note: " + title, Toast.LENGTH_SHORT).show()
        }
    }
}
